include/boost/geometry/strategies/cartesian/side_of_intersection.hpp

```diff
@@ -27,6 +35,104 @@ namespace boost { namespace geometry
 namespace strategy { namespace side
 {
 
+namespace detail
+{
+
+template <typename T, T N, std::size_t P>
+struct integral_pow
+{
+    static const T value = N * integral_pow<T, N, P-1>::value;
+};
+
+template <typename T, T N>
+struct integral_pow<T, N, 0>
+{
+    static const T value = 1;
+};
+
+// A tool for multiplication of integers avoiding overflow
+// It's a temporary workaround until we can use Multiprecision
+template <typename T>
+struct multiplicable_integral
+{
+    // This tool can't be used for non-integral coordinate types
+    // also sign_of_product and sign_of_compare functions must be modified
+    // (comparsons and multiplication)
+    BOOST_STATIC_ASSERT(boost::is_integral<T>::value);
+
+    static const std::size_t bits = CHAR_BIT * sizeof(T);
+    static const std::size_t half_bits = bits / 2;
+    typedef typename boost::make_unsigned<T>::type unsigned_type;
+    static const unsigned_type half_base
+        = integral_pow<unsigned_type, 2, half_bits>::value;
+
+    int sign;
+    unsigned_type ms;
+    unsigned_type ls;
+
+    multiplicable_integral(int sign_, unsigned_type ms_, unsigned_type ls_)
+        : sign(sign_), ms(ms_), ls(ls_)
+    {}
+
+    explicit multiplicable_integral(T const& val)
+    {
+        unsigned_type val_u = val > 0 ?
+                                unsigned_type(val)
+                              : val == (std::numeric_limits<T>::min)() ?
+                                  unsigned_type((std::numeric_limits<T>::max)()) + 1
```

Comment:
Is that part of the boost coding rules? I know that `std::min` and `std::max` need to go inside parentheses, but this is the same here.